### PR TITLE
Phone verification update

### PR DESCRIPTION
**Pull request by kpbacode**

### CODE REVIEW

File: app/controllers/kaui/accounts_controller.rb [hunk 1]
```diff
@@ -54,8 +54,18 @@ def new
   end
 
   def create
+
     @account = Kaui::Account.new(params.require(:account).delete_if { |key, value| value.blank? })
 
+    ap @account
```
Comment (pierre) [inline app/controllers/kaui/accounts_controller.rb:60]: Leftover debug :-)

File: app/models/kaui/account_validate.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# class KauiPhoneValid
```
Comment (pierre) [inline app/models/kaui/account_validate.rb:1]: Can we get rid of this file now?

File: app/controllers/kaui/accounts_controller.rb (hunk 1 above)
Comment (kpbacode) [reply app/controllers/kaui/accounts_controller.rb]: Done. Sorry

File: app/models/kaui/account_validate.rb (hunk 2 above)
Comment (kpbacode) [reply app/models/kaui/account_validate.rb]: Done. Sorry

File: app/views/kaui/layouts/kaui_flash.html.erb [hunk 3]
```diff
@@ -8,6 +8,30 @@
         </div>
     <% end %>
 <% end %>
+
+
+
+<% if @account && @account.errors.any? %>
```
Comment (pierre) [inline app/views/kaui/layouts/kaui_flash.html.erb:14]: FYI I think this is breaking the tests: https://github.com/killbill/killbill-admin-ui/actions/runs/3068383150/jobs/4956276626

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: Maybe this will fix

Comment (pierre) [reply app/views/kaui/layouts/kaui_flash.html.erb]: https://github.com/killbill/killbill-admin-ui/actions/runs/3069706935/jobs/4958948909

Are you able to run the tests locally?

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: 1. Can we update documentation for standalone tomcat deployment?
https://docs.killbill.io/latest/userguide_configuration.html#global_configuration_properties

a) create tomcat/bin/setenv.sh  file with content
`export CATALINA_OPTS="$CATALINA_OPTS -Dorg.killbill.server.test.mode=true"`

Now I was able to run tests. I will investigate them.

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: `/Users/mark/github/killbill-admin-ui/test/functional/kaui/functional_test_helper_nosetup.rb:68: warning: The called method `get' is defined here
...F

Failure:
Kaui::AccountEmailsControllerTest#test_should_handle_Kill_Bill_errors_during_deletion [/Users/mark/github/killbill-admin-ui/test/functional/kaui/account_emails_controller_test.rb:34]:
--- expected
+++ actual
@@ -1 +1 @@
-"Error while communicating with the Kill Bill server: "
+"Error while communicating with the Kill Bill server: <!doctype html><html lang=\"en\"><head><title>HTTP Status 404 – Not Found</title><style type=\"text/css\">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 404 – Not Found</h1><hr class=\"line\" /><p><b>Type</b> Status Report</p><p><b>Message</b> Not Found</p><p><b>Description</b> The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.</p><hr class=\"line\" /><h3>Apache Tomcat/8.5.82</h3></body></html>"`

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: Failure:
Kaui::AccountTagsControllerTest#test_should_handle_Kill_Bill_errors_when_getting_edit_screen [/Users/mark/github/killbill-admin-ui/test/functional/kaui/account_tags_controller_test.rb:9]:
--- expected
+++ actual
@@ -1 +1 @@
-"Error while communicating with the Kill Bill server: "
+"Error while communicating with the Kill Bill server: <!doctype html><html lang=\"en\"><head><title>HTTP Status 404 – Not Found</title><style type=\"text/css\">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 404 – Not Found</h1><hr class=\"line\" /><p><b>Type</b> Status Report</p><p><b>Message</b> Not Found</p><p><b>Description</b> The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.</p><hr class=\"line\" /><h3>Apache Tomcat/8.5.82</h3></body></html>"

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: Error:
Kaui::AccountsControllerTest#test_should_create_account:
ActionController::UrlGenerationError: No route matches {:account_id=>nil, :action=>"show", :controller=>"kaui/accounts"}, possible unmatched constraints: [:account_id]
    actionpack (5.2.8.1) lib/action_dispatch/journey/formatter.rb:57:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:744:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:775:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:822:in `url_for'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:273:in `call'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:214:in `call'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:331:in `block (2 levels) in define_url_helper'
    actionpack (5.2.8.1) lib/action_dispatch/testing/assertions/routing.rb:174:in `method_missing'
    /Users/mark/github/killbill-admin-ui/test/functional/kaui/accounts_controller_test.rb:121:in `block in <class:AccountsControllerTest>'
    minitest (5.16.3) lib/minitest/test.rb:98:in `block (3 levels) in run'
    minitest (5.16.3) lib/minitest/test.rb:195:in `capture_exceptions'
    minitest (5.16.3) lib/minitest/test.rb:95:in `block (2 levels) in run'
    minitest (5.16.3) lib/minitest.rb:296:in `time_it'
    minitest (5.16.3) lib/minitest/test.rb:94:in `block in run'
    minitest (5.16.3) lib/minitest.rb:391:in `on_signal'
    minitest (5.16.3) lib/minitest/test.rb:243:in `with_info_handler'
    minitest (5.16.3) lib/minitest/test.rb:93:in `run'
    minitest (5.16.3) lib/minitest.rb:1059:in `run_one_method'
    minitest (5.16.3) lib/minitest.rb:365:in `run_one_method'
    minitest (5.16.3) lib/minitest.rb:352:in `block (2 levels) in run'
    minitest (5.16.3) lib/minitest.rb:351:in `each'
    minitest (5.16.3) lib/minitest.rb:351:in `block in run'
    minitest (5.16.3) lib/minitest.rb:391:in `on_signal'
    minitest (5.16.3) lib/minitest.rb:378:in `with_info_handler'
    minitest (5.16.3) lib/minitest.rb:350:in `run'
    railties (5.2.8.1) lib/rails/test_unit/line_filtering.rb:10:in `run'
    minitest (5.16.3) lib/minitest.rb:182:in `block in __run'
    minitest (5.16.3) lib/minitest.rb:182:in `map'
    minitest (5.16.3) lib/minitest.rb:182:in `__run'
    minitest (5.16.3) lib/minitest.rb:159:in `run'
    minitest (5.16.3) lib/minitest.rb:83:in `block in autorun'

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: Error:
Kaui::AccountsControllerTest#test_should_validate_external_key_if_found:
ActionController::UrlGenerationError: No route matches {:account_id=>nil, :action=>"show", :controller=>"kaui/accounts"}, possible unmatched constraints: [:account_id]
    actionpack (5.2.8.1) lib/action_dispatch/journey/formatter.rb:57:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:744:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:775:in `generate'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:822:in `url_for'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:273:in `call'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:214:in `call'
    actionpack (5.2.8.1) lib/action_dispatch/routing/route_set.rb:331:in `block (2 levels) in define_url_helper'
    actionpack (5.2.8.1) lib/action_dispatch/testing/assertions/routing.rb:174:in `method_missing'
    /Users/mark/github/killbill-admin-ui/test/functional/kaui/accounts_controller_test.rb:225:in `block in <class:AccountsControllerTest>'
    minitest (5.16.3) lib/minitest/test.rb:98:in `block (3 levels) in run'
    minitest (5.16.3) lib/minitest/test.rb:195:in `capture_exceptions'
    minitest (5.16.3) lib/minitest/test.rb:95:in `block (2 levels) in run'
    minitest (5.16.3) lib/minitest.rb:296:in `time_it'
    minitest (5.16.3) lib/minitest/test.rb:94:in `block in run'
    minitest (5.16.3) lib/minitest.rb:391:in `on_signal'
    minitest (5.16.3) lib/minitest/test.rb:243:in `with_info_handler'
    minitest (5.16.3) lib/minitest/test.rb:93:in `run'
    minitest (5.16.3) lib/minitest.rb:1059:in `run_one_method'
    minitest (5.16.3) lib/minitest.rb:365:in `run_one_method'
    minitest (5.16.3) lib/minitest.rb:352:in `block (2 levels) in run'
    minitest (5.16.3) lib/minitest.rb:351:in `each'
    minitest (5.16.3) lib/minitest.rb:351:in `block in run'
    minitest (5.16.3) lib/minitest.rb:391:in `on_signal'
    minitest (5.16.3) lib/minitest.rb:378:in `with_info_handler'
    minitest (5.16.3) lib/minitest.rb:350:in `run'
    railties (5.2.8.1) lib/rails/test_unit/line_filtering.rb:10:in `run'
    minitest (5.16.3) lib/minitest.rb:182:in `block in __run'
    minitest (5.16.3) lib/minitest.rb:182:in `map'
    minitest (5.16.3) lib/minitest.rb:182:in `__run'
    minitest (5.16.3) lib/minitest.rb:159:in `run'
    minitest (5.16.3) lib/minitest.rb:83:in `block in autorun'

Comment (kpbacode) [reply app/views/kaui/layouts/kaui_flash.html.erb]: Fixed - https://github.com/killbill/killbill-admin-ui/pull/313#discussion_r973581895

File: test/dummy/db/schema.rb [hunk 4]
```diff
@@ -11,42 +10,41 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20150112232813) do
+ActiveRecord::Schema.define(version: 2015_12_10_120915) do
```
Comment (pierre) [inline test/dummy/db/schema.rb:13]: BTW, once the tests are fixed, can you get rid of these changes as well as the new files (test/dummy/config/database_dev.yml, test/dummy/config/database_github.yml, etc.), which shouldn't be needed?

Would also be great if you could clean-up the Git history (new PR might be easiest).

Thanks!

Comment (kpbacode) [reply test/dummy/db/schema.rb:13]: > BTW, once the tests are fixed, can you get rid of these changes as well as the new files (test/dummy/config/database_dev.yml, test/dummy/config/database_github.yml, etc.), which shouldn't be needed?
> 
> Would also be great if you could clean-up the Git history (new PR might be easiest).
> 
> Thanks!

Will be done.

Comment (kpbacode) [reply test/dummy/db/schema.rb:13]: killbill-client-ruby is not active record model, so ActiveModel::Validation will not work.

Comment (kpbacode) [reply test/dummy/db/schema.rb:13]: Running tests

**Pull request closed without merging**